src/pages/Editor.vue

```diff
@@ -602,35 +666,169 @@ export default {
       this.dialogTitle = this.publishDialogTitle;
       this.dialogDescription = this.publishDialogDescription;
       this.dialogConfirmButtonConfig = {
+        enabled: true,
         text: "Yes",
         class:
           "bg-primary-button hover:bg-primary-button-hover focus:outline-none focus:ring-0",
       };
       this.dialogCancelButtonConfig = {
+        enabled: true,
         text: "No",
         class: "bg-white hover:bg-gray-100 focus:outline-none text-primary",
       };
+      // closing the dialog executes this action
+      this.dialogAction = "publish"
       // show the dialogue
       this.showDialogBox = true;
     },
     dialogConfirmed() {
       // invoked when the confirm button of the dialog box is clicked
       // update the dialog properties
       this.dialogConfirmButtonConfig = {
+        enabled: true,
         class: "hidden",
       };
       this.dialogCancelButtonConfig = {
+        enabled: true,
         class: "hidden",
       };
       this.dialogDescription = "";
-      this.dialogTitle = this.publishInProgressDialogTitle;
-      // publish the plio or its changes
-      this.publishPlio();
+
+      // call separate methods depening on the dialog action that
+      // was set
+      if (this.dialogAction == "publish")
+        this.dialogPublish()
+      else if (this.dialogAction == "deleteItem")
+        this.dialogDeleteItem()
+      else if (this.dialogAction == "closeDialog")
+        this.showDialogBox = false
+
+      // reset the dialog action value
+      this.dialogAction = ""
     },
     dialogCancelled() {
       // invoked when the cancel button of the dialog box is clicked
       this.showDialogBox = false;
     },
+    showCannotAddItemDialog() {
+      // set up the dialog properties when user tries to add an item
+      // at an invalid time
+      this.dialogTitle = "Cannot add a new question here";
+      this.dialogDescription = "Questions should be at least 2 seconds apart. Please choose a different time for the question";
+      this.dialogConfirmButtonConfig = {
+        enabled: true,
+        text: "Got it!",
+        class: "bg-primary-button hover:bg-primary-button-hover focus:outline-none focus:ring-0",
+      };
+      this.dialogCancelButtonConfig = {
+        enabled: false,
+        text: "",
+        class: ""
+      };
+
+      // carry out the closeDialog action when dialog is closed
+      this.dialogAction = "closeDialog"
+      // show the dialogue
+      this.showDialogBox = true;
+    },
+    dialogPublish() {
+      this.dialogTitle = this.publishInProgressDialogTitle;
+      // publish the plio or its changes
+      this.publishPlio();
+    },
+    dialogDeleteItem() {
+      this.dialogTitle = this.deleteItemInProgressDialogTitle;
+      // delete the selected item after user confirms
+      this.deleteSelectedItem();
+    },
```

Comment:
these names seem more appropriate - warna I thought this functions will open the dialog box for these dialog actions.
```suggestion
    confirmPublish() {
      this.dialogTitle = this.publishInProgressDialogTitle;
      // publish the plio or its changes
      this.publishPlio();
    },
    confirmDeleteItem() {
      this.dialogTitle = this.deleteItemInProgressDialogTitle;
      // delete the selected item after user confirms
      this.deleteSelectedItem();
    },
```